src/analysis/box_integral/partition/tagged.lean

```diff
@@ -0,0 +1,369 @@
+/-
+Copyright (c) 2021 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import analysis.box_integral.partition.basic
+
+/-!
+# Tagged partitions
+
+A tagged (pre)partition is a (pre)partition `π` enriched with a tagged point for each box of
+‵π`. For simplicity we require that the function `box_integral.tagged_prepartition.tag` is defined
+on all boxes `J : box ι` but use its values only on boxes of the partition. Given `π :
+box_integral.tagged_partition I`, we require that each `box_integral.tagged_partition π J` belongs
+to `box_integral.box.Icc I`. If for every `J ∈ π`, `π.tag J` belongs to `J.Icc`, then `π` is called
+a *Henstock* partition. We do not include this assumption into the definition of a tagged
+(pre)partition because McShane integral is defined as a limit along tagged partitions without this
+requirement.
+
+### Tags
+
+rectangular box, box partition
+-/
+
+noncomputable theory
+open_locale classical ennreal nnreal
+open set function
+
+namespace box_integral
+
+variables {ι : Type*}
+
+/-- A tagged prepartition is a prepartition enriched with a tagged point for each box of the
+prepartition. -/
```

Comment:
Can you mention here what you already said in the file-level docstring, that `tag` is defined for all boxes for simplicity, but that you will only use it over boxes of the prepartition.